apps/site/src/app/apply/page.tsx

```diff
@@ -25,6 +25,14 @@ export default async function Apply({
 		redirect("/portal");
 	}
 
+	if (
+		searchParams !== undefined &&
+		searchParams?.prefaceAccepted === "true" &&
+		identity.uid === null
+	) {
+		redirect("/login");
+	}
+
 	const applyBody =
 		searchParams !== undefined &&
 		searchParams?.prefaceAccepted === "true" ? (
```

Comment:
Chore: We're repeating the same conditional logic for checking the `prefaceAccepted` state. Could we extract this into a variable we can reference in both spots?